DNN Platform/Modules/Journal/ServicesController.cs

```diff
@@ -1,619 +1,621 @@
+        public HttpResponseMessage GetSuggestions(string keyword)
+        {
+            try
+            {
+                var findedUsers = new List<SuggestDTO>();
+                var relations = RelationshipController.Instance.GetUserRelationships(this.UserInfo);
+                foreach (var ur in relations)
+                {
+                    var targetUserId = ur.UserId == this.UserInfo.UserID ? ur.RelatedUserId : ur.UserId;
+                    var targetUser = UserController.GetUserById(this.PortalSettings.PortalId, targetUserId);
+                    var relationship = RelationshipController.Instance.GetRelationship(ur.RelationshipId);
+                    if (ur.Status == RelationshipStatus.Accepted && targetUser != null
+                        && ((relationship.RelationshipTypeId == (int)DefaultRelationshipTypes.Followers && ur.RelatedUserId == this.UserInfo.UserID)
+                                || relationship.RelationshipTypeId == (int)DefaultRelationshipTypes.Friends)
+                        && (targetUser.DisplayName.ToLowerInvariant().Contains(keyword.ToLowerInvariant())
+                                || targetUser.DisplayName.ToLowerInvariant().Contains(keyword.Replace("-", " ").ToLowerInvariant()))
+                        && findedUsers.All(s => s.userId != targetUser.UserID))
+                    {
+                        findedUsers.Add(new SuggestDTO
+                        {
+                            displayName = targetUser.DisplayName.Replace(" ", "-"),
+                            userId = targetUser.UserID,
+                            avatar = targetUser.Profile.PhotoURL,
+                            key = keyword,
+                        });
+                    }
+                }
+
+                return this.Request.CreateResponse(HttpStatusCode.OK, findedUsers.Cast<object>().Take(5));
+            }
+            catch (Exception exc)
+            {
+                Logger.Error(exc);
+                return this.Request.CreateErrorResponse(HttpStatusCode.InternalServerError, exc);
+            }
+        }
+
+        private static bool IsImageFile(string relativePath)
+        {
+            if (relativePath == null)
+            {
+                return false;
+            }
+
+            if (relativePath.Contains("?"))
+            {
+                relativePath = relativePath.Substring(
+                    0,
+                    relativePath.IndexOf("?", StringComparison.InvariantCultureIgnoreCase));
+            }
+
+            var extension = relativePath.Substring(relativePath.LastIndexOf(
+                ".",
+                StringComparison.Ordinal) + 1).ToLowerInvariant();
+            return AcceptedFileExtensions.Contains(extension);
+        }
+
+        private static bool IsAllowedLink(string url)
+        {
+            return !string.IsNullOrEmpty(url) && (url.Contains("https://") || url.Contains("http://") || url.Contains("www."));

```

Comment:
I would do this instead to support all different types of URLs (except file, which would be local to the client).

```suggestion
            Uri uriResult;
            return Uri.TryCreate(url, UriKind.Absolute, out uriResult)
                && uriResult.Scheme != Uri.UriSchemeFile;
```

Reply:
Great suggestion @valadas 🎉 🌮

Reply:
If other reviewers have concerns with allowing too many schemes, then this can also be rewritten to allow just what we want instead of not allowing what we want (reverse the logic)

Reply:
I am all for improving it, I figured someone was going to toss us the one up here if nothing else let this fly possibly as is.  I am going to test this solution now.  Thanks for all the quick responses.  I would love to put this one to bed!